src/cli/tests/test_filter_fixtures.py

```diff
@@ -0,0 +1,786 @@
+        ]
+
+        if len(available_ordered) < 2:
+            pytest.skip("Auto-setup requires at least 2 chronologically ordered forks")
+
+        second_latest_fork = available_ordered[-2]
+
+        # Create filtered "after" directory
+        after_fixtures_dir = temp_output_dir
+
+        runner = CliRunner()
+        result = runner.invoke(
+            filter_fixtures,
+            [
+                "--input",
+                str(before_fixtures_dir),
+                "--output",
+                str(after_fixtures_dir),
+                "--until",
+                second_latest_fork,
+                "--quiet",
+            ],
+        )
+
+        if result.exit_code != 0:
+            pytest.fail(f"Auto-setup filter_fixtures failed: {result.output}")
+
+        print(f"   Before: {before_fixtures_dir.name}/ (all forks)")
+        print(f"   After:  {after_fixtures_dir.name}/ (filtered --until {second_latest_fork})")
+        print()
+
+    if not before_fixtures_dir.exists():
+        pytest.fail(
+            f"❌ Before fixtures directory does not exist: {before_fixtures_dir}\n\n"
+            "Ensure the development fixtures directory exists and contains fixtures.\n"
+            "This directory should contain the output from:\n"
+            "  fill --output=fixtures_develop --until=ForkN+1"
+        )
+
+    if not after_fixtures_dir.exists():
+        pytest.fail(
+            f"❌ After fixtures directory does not exist: {after_fixtures_dir}\n\n"
+            "Ensure the stable fixtures directory exists and contains filtered fixtures.\n"
+            "This directory should contain the output from:\n"
+            "  filter_fixtures --input=fixtures_develop --output=fixtures_stable --until=ForkN"
+        )
+
+    # Load index.json from both directories
+    before_index_path = before_fixtures_dir / ".meta" / "index.json"
+    after_index_path = after_fixtures_dir / ".meta" / "index.json"
+
+    if not before_index_path.exists():
+        pytest.fail(
+            f"❌ Before index.json not found: {before_index_path}\n\n"
+            "The fixtures directory appears to be incomplete or corrupted.\n"
+            "Ensure the directory was generated properly with 'fill' command and contains:\n"
+            "  - .meta/index.json\n"
+            "  - fixture files in subdirectories"
+        )
+
+    if not after_index_path.exists():
+        pytest.fail(
+            f"❌ After index.json not found: {after_index_path}\n\n"
+            "The filtered fixtures directory appears to be incomplete or corrupted.\n"
+            "Ensure the directory was generated properly with 'filter_fixtures' and contains:\n"
+            "  - .meta/index.json\n"
+            "  - filtered fixture files in subdirectories"
+        )
+
+    with before_index_path.open() as f:
+        before_index = json.load(f)
+
+    with after_index_path.open() as f:
+        after_index = json.load(f)
```

Comment:
This test function is critical to verify releases in CI: It should be ran (see docstring/PR description) to compare the original (fixtures_develop) and filtered output (fixtures_stable) to automatically sanity check them and provide console output for an easy manual visual check.

Note, this is only validating via index files. `filter_fixtures` will generate a new index.json for the new fixtures output dir via `genindex`. But, in a way, the verification here is a "light" verification. Perhaps we should do a full verification by loading all fixture files ourselves here, but perhaps not, I mean this is what `genindex` does anyway... Will leave as-is for now. Happy to hear your feedback.